stdlib/ballerina-builtin/src/main/java/org/ballerinalang/nativeimpl/builtin/systemlib/Sprintf.java

```diff
@@ -0,0 +1,113 @@
+package org.ballerinalang.nativeimpl.builtin.systemlib;
+
+import org.ballerinalang.bre.Context;
+import org.ballerinalang.model.types.TypeKind;
+import org.ballerinalang.model.values.BRefValueArray;
+import org.ballerinalang.model.values.BString;
+import org.ballerinalang.model.values.BValue;
+import org.ballerinalang.natives.AbstractNativeFunction;
+import org.ballerinalang.natives.annotations.Argument;
+import org.ballerinalang.natives.annotations.BallerinaFunction;
+import org.ballerinalang.natives.annotations.ReturnType;
+import org.ballerinalang.util.exceptions.BLangExceptionHelper;
+import org.ballerinalang.util.exceptions.RuntimeErrors;
+
+@BallerinaFunction(
+        packageName = "ballerina.builtin",
+        functionName = "sprintf",
+        args = {@Argument(name = "format", type = TypeKind.STRING),
+                @Argument(name = "args", type = TypeKind.ARRAY)},
+        returnType = {@ReturnType(type = TypeKind.STRING)},
+        isPublic = true
+)
+
+public class Sprintf extends AbstractNativeFunction {
+    @Override
+    public BValue[] execute(Context context) {
+        String format = getStringArgument(context, 0);
+        BRefValueArray args = (BRefValueArray) getRefArgument(context, 0);
+        StringBuilder result = new StringBuilder();
+
+        /* Special chars in case additional formatting is required later
+         *
+         * Primitive built-in types (Same as Java String.format())
+         * b            boolean
+         * d            int
+         * f            float
+         * s            string
+         * x            hex
+         * X            HEX (ALL_CAPS)
+         * o            octal
+         * B            binary
+         *
+         * Complex built-in types
+         * m            message
+         * e            exception
+         * p            map
+         * l            xml
+         * L            xmlDocument
+         * j            json
+         * t            datatable
+         *
+         * User defined types
+         * r            struct
+         * a            arrays
+         */
+
+        // i keeps index for checking %
+        // j reads format specifier to apply
+        // k records number of format specifiers seen so far, used to read respective array element
+        for (int i = 0, j = 0, k = 0; i < format.length(); i++) {
+            if (format.charAt(i) == '%' && ((i+1) < format.length())) {
+
+                // skip % character
+                j = i + 1;
+
+                if (k >= args.size()) {
+                    // there's not enough arguments
+                    throw BLangExceptionHelper.getRuntimeException(
+                            RuntimeErrors.NOT_ENOUGH_FORMAT_ARGUMENTS);
```

Comment:
I think this line can be merged with the above line.

Reply:
Fixed

Reply:
Still `throw BLangExceptionHelper.getRuntimeException(` and  `RuntimeErrors.NOT_ENOUGH_FORMAT_ARGUMENTS);` is in two lines. You might have forgotten to push the commit.

Reply:
Yup, fixed now. I read your comment wrong.